lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -1614,6 +1621,459 @@ class ConvertAtenAdaptivePoolOp : public OpConversionPattern<OpTy> {
     return success();
   }
 };
+
+struct ConvertRoiAlignOp final
+    : OpConversionPattern<Torch::TorchvisionRoiAlignOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  static SmallVector<Value>
+  coordinateTransform(OpBuilder &b, Torch::TorchvisionRoiAlignOp op,
+                      Location loc, SmallVector<Value> outputSizes, Value input,
+                      SmallVector<Value> inputSizes,
+                      SmallVector<Value> scaleValues, std::string coordStr,
+                      bool alignCornersBool, SmallVector<Value> indices,
+                      bool clip) {
+
+    unsigned dimOffset = 2;
+    auto inputType = cast<RankedTensorType>(input.getType());
+    auto inputRank = inputType.getRank();
+
+    Value cstOneFloat =
+        b.create<arith::ConstantOp>(loc, b.getF32FloatAttr(1.0));
+    Value cstHalf = b.create<arith::ConstantOp>(loc, b.getF32FloatAttr(0.5));
+    Value zero = b.create<arith::ConstantOp>(loc, b.getF32FloatAttr(0.0));
+
+    SmallVector<Value> proj;
+    for (unsigned i = 0; i < inputRank - dimOffset; i++) {
+      // length_original
+      Value inputFP =
+          b.create<arith::SIToFPOp>(loc, b.getF32Type(), inputSizes[i]);
+      // length_resized
+      Value outputSizeFP =
+          b.create<arith::SIToFPOp>(loc, b.getF32Type(), outputSizes[i]);
+      // scale = length_resized/length_original
+      Value scale;
+
+      if (alignCornersBool) {
+        // x_original = x_resized * (length_original - 1) / (length_resized - 1)
+        Value inputSubOne = b.create<arith::SubFOp>(loc, inputFP, cstOneFloat);
+        Value outputSizeSubOne =
+            b.create<arith::SubFOp>(loc, outputSizeFP, cstOneFloat);
+        Value cmp = b.create<arith::CmpFOp>(loc, arith::CmpFPredicate::UEQ,
+                                            outputSizeSubOne, zero);
+        scale = b.create<arith::DivFOp>(loc, inputSubOne, outputSizeSubOne);
+        scale = b.create<arith::SelectOp>(loc, cmp, zero, scale);
+        coordStr = "_align_corners";
+      } else if (scaleValues.empty())
+        scale = b.create<arith::DivFOp>(loc, outputSizeFP, inputFP);
+      else
+        scale = scaleValues[i];
```

Comment:
When one path of an `if` statement is surrounded with braces, make sure that all the other ones also get braces